src/test/perf_counters.cc

```diff
@@ -219,3 +223,54 @@ TEST(PerfCounters, ResetPerfCounters) {
   ASSERT_EQ("", client.do_request("{ \"prefix\": \"perf reset\", \"var\": \"test_perfcounter_1\", \"format\": \"json\" }", &msg));
   ASSERT_EQ(sd("{\"error\":\"Not find: test_perfcounter_1\"}"), msg);
 }
+
+enum {
+  TEST_PERFCOUNTERS3_ELEMENT_FIRST = 800,
+  TEST_PERFCOUNTERS3_ELEMENT_1,
+  TEST_PERFCOUNTERS3_ELEMENT_2,
+  TEST_PERFCOUNTERS3_ELEMENT_LAST,
+};
+
+static PerfCounters* setup_test_perfcounters3(CephContext *cct)
+{
+  PerfCountersBuilder bld(cct, "test_perfcounter_3",
+          TEST_PERFCOUNTERS3_ELEMENT_FIRST, TEST_PERFCOUNTERS3_ELEMENT_LAST);
+  bld.add_time_avg(TEST_PERFCOUNTERS3_ELEMENT_1, "ceph::timespan");
+  bld.add_time_avg(TEST_PERFCOUNTERS3_ELEMENT_2, "utime_t");
+  return bld.create_perf_counters();
+}
+
+TEST(CoarseSteadyClock, PerfCounter) {
+  PerfCountersCollection *coll = g_ceph_context->get_perfcounters_collection();
+  PerfCounters* fake_pf = setup_test_perfcounters3(g_ceph_context);
```

Comment:
`fake_pf` is leaked. should remove it from `coll`, then delete it. i'd suggest manage its life cycle using a `unique_ptr`.

Reply:
Resolved